File: folkmq-lang-js/folkmq-test/demo02-node-ws/_namespace.js
const {FolkMQ} = require("@noear/folkmq");

async function main() {
    const client =  await FolkMQ.createClient("folkmq:ws://127.0.0.1:18602")
        .nameAs("demoapp")
        .namespaceAs("DEFAULT")
        .connect();

    //订阅
    client.subscribe("test.topic", null, true, message => {
        console.log(message.getBodyAsString());
    });

    //发布
    for (let i = 0; i < 10; i++) {
        client.publish("test.topic", FolkMQ.newMessage("hot-" + i));
    }
}

main();

File: folkmq-lang-python/folkmq/folkmq-test/demo02/_namespace.py
import asyncio

from socketd.utils.LogConfig import log

from folkmq.FolkMQ import FolkMQ
from folkmq.client.MqMessage import MqMessage


async def main():
    client = await (FolkMQ.create_client("folkmq:ws://127.0.0.1:18602")
                    .name_as("demoapp")
                    .namespace_as("DEFAULT")
                    .connect())

    # 订阅
    await client.subscribe("test.topic", None, True, lambda message:
        log.info(message.get_body_as_string())
    )

    # 发布
    for i in range(10):
        await client.publish("test.topic", MqMessage("hot-" + str(i)))

    await asyncio.Future()

if __name__ == "__main__":
    asyncio.run(main())

